beater/handlers.go

```diff
@@ -52,17 +52,63 @@ type routeMapping struct {
 	ProcessorFactory
 }
 
+type serverResponse struct {
+	err     error
+	code    int
+	counter *monitoring.Int
+}
+
 var (
-	serverMetrics  = monitoring.Default.NewRegistry("apm-server.server")
-	requestCounter = monitoring.NewInt(serverMetrics, "requests.counter")
-	responseValid  = monitoring.NewInt(serverMetrics, "response.valid")
-	responseErrors = monitoring.NewInt(serverMetrics, "response.errors")
+	serverMetrics = monitoring.Default.NewRegistry("apm-server.server")
+	counter       = func(s string) *monitoring.Int {
+		return monitoring.NewInt(serverMetrics, s)
+	}
+	requestCounter = counter("request.count")
+	responseCount  = counter("response.count")
+	responseErrors = counter("response.errors.count")
```

Comment:
Now that you split up `response.valid` I'd also expect a `responseSucesses` or removing the `responseErrors`